### ISSUE TITLE
currentUser.displayName

### ISSUE THREAD

**Issue by theevo** [coronacation/MVP#23]
## tl;dr 

We basically need to fill this property with some data. Having this field available (and not nil) will make it very easy for the chat app to display names.

## displayName is nil?

[Email authentication](https://firebase.google.com/docs/auth/users) does not populate it, therefore it returns nil.

> If the user signed up with an email address and password, only the primary email address property is populated

## Possible Solutions

1. **We could ask the user**. Add a textfield at signup. Make sure it's clear this will be their name in correspondence. Maybe we can auto-populate it with their list name?
2. **We could create it behind the scenes** for the user by taking some variation of Firstname + Last initial (Theo V.) Another possibility: take the first part of their email address (everything before the @)

## Implementation code

The exact method call can be found in [Firebase documentation](https://firebase.google.com/docs/auth/ios/manage-users#update_a_users_profile).

```
let changeRequest = Auth.auth().currentUser?.createProfileChangeRequest()
changeRequest?.displayName = displayName
changeRequest?.commitChanges { (error) in
  // ...
}
```

## Implications

We might have to think about adding displayName as an editable field in the Profile section.

**Comment by theevo** [comment]
Closing this issue. It's no longer relevant to our app since the designers will reveal people's actual first names and last names. There will be no aliases or usernames. Just first name and last name.

Sorry for the drama!
